src/query/service/src/pipelines/processors/transforms/hash_join/probe_join/inner_join.rs

```diff
@@ -62,17 +72,25 @@ impl JoinHashTable {
             if let Some(v) = probe_result_ptr {
                 let probed_rows = v.get();
 
-                if probe_indexes.len() + probed_rows.len() < probe_indexes.capacity() {
-                    build_indexes.extend_from_slice(probed_rows);
-                    probe_indexes.extend(repeat(i as u32).take(probed_rows.len()));
+                if probed_num + probed_rows.len() < JOIN_MAX_BLOCK_SIZE {
+                    for it in probed_rows {
+                        build_indexes.push(it);
+                    }
+                    probe_indexes[probe_indexes_len] = (i as u32, probed_rows.len() as u32);
+                    probe_indexes_len += 1;
```

Comment:
How about using `HashMap<u32, u32>` to replace `Vec<(u32, u32)>`, then you don't need `probe_indexes_len`

Reply:
we need `probe_indexes_len` to reuse the `probe_indexes`.